storage/api/csk_operations.py

```diff
@@ -0,0 +1,160 @@
+  #     https://developers.google.com/api-client-library/python/apis/
+  return discovery.build("storage", "v1", credentials=credentials)
+
+
+def upload_object(bucket, filename, encryption_key, key_hash):
+  """Uploads an object, specifying a custom encryption key."""
+  service = create_service()
+
+  with open(filename, "rb") as f:
+    request = service.objects().insert(
+        bucket=bucket, name=filename,
+        # You can also just set media_body=filename, but for the sake of
+        # demonstration, pass in the more generic file handle, which could
+        # very well be a StringIO or similar.
+        media_body=http.MediaIoBaseUpload(f, "application/octet-stream"))
+    request.headers["x-goog-encryption-algorithm"] = "AES256"
+    request.headers["x-goog-encryption-key"] = encryption_key
+    request.headers["x-goog-encryption-key-sha256"] = key_hash
+
+    resp = request.execute()
+
+  return resp
+
+
+def download_object(bucket, obj, out_file, encryption_key, key_hash):
+  """Downloads an object protected by a custom encryption key."""
+  service = create_service()
+
+  request = service.objects().get_media(bucket=bucket, object=obj)
+  request.headers["x-goog-encryption-algorithm"] = "AES256"
+  request.headers["x-goog-encryption-key"] = encryption_key
+  request.headers["x-goog-encryption-key-sha256"] = key_hash
+
+  # Unfortunately, http.MediaIoBaseDownload overwrites HTTP headers,
+  # and so it cannot be used here. Instead, we shall download as a
+  # single request.
+  out_file.write(request.execute())
+
+
+def rotate_key(bucket, obj, current_encryption_key, current_key_hash,
+               new_encryption_key, new_key_hash):
+  """Changes the encryption key used to store an existing object."""
+  service = create_service()
+
+  request = service.objects().rewrite(
+      sourceBucket=bucket, sourceObject=obj,
+      destinationBucket=bucket, destinationObject=obj,
+      body={})
+
+  # For very large objects, calls to rewrite may not complete on the first
+  # call and may need to be resumed.
+  while True:
+    request.headers["x-goog-copy-source-encryption-algorithm"] = "AES256"
+    request.headers["x-goog-copy-source-encryption-key"] = current_encryption_key
+    request.headers["x-goog-copy-source-encryption-key-sha256"] = current_key_hash
+    request.headers["x-goog-encryption-algorithm"] = "AES256"
+    request.headers["x-goog-encryption-key"] = new_encryption_key
+    request.headers["x-goog-encryption-key-sha256"] = new_key_hash
+
+    rewrite_response = request.execute()
+
+    if rewrite_response["done"]:
+      break
+
+    print "Continuing rewrite call..."
+    request = service.objects().rewrite(
+        source_bucket=bucket, source_object=obj,
+        destination_bucket=bucket, destination_object=obj,
+        rewriteToken=rewrite_response["rewriteToken"])
+    rewrite_response.execute()
+
+
+def main(bucket, filename):
+  print "Uploading object gs://%s/%s" % (bucket, filename)
```

Comment:
please use print as a function - this sample should work equally well on Python 3.